Recruit-Training/Embedded-Recruit-Training/Core/Src/main.c

```diff
@@ -328,41 +328,42 @@ void HAL_CAN_RxFifo0MsgPendingCallback(CAN_HandleTypeDef* hcan)
     }
 
     //TODO: Match StdId of header and data length content for green and blue messages and check data for set bit of toggling green and blue led
-    if ((hdr.StdId == BLUE_MESSAGE_STDID || hdr.StdId == GREEN_MESSAGE_STDID) && hdr.DLC == 1)
-    {
-        if (data[0] == 1 && data[3] == 1 && data[7] == 1)
+    
+
+        if (hdr.StdId == BLUE_MESSAGE_STDID && hdr.DLC == 1)
         {
-            if (BlueToggleFlag)
-                BlueToggleFlag = 0;
-            else
+            uint8_t mask = 0b10001001
+            if (data[0] & mask == mask)
+            {
                 BlueToggleFlag = 1;
-
+            }
+            else 
+            {
+                BlueToggleFlag = 0;
+            }
         }
-    
-        if (data[0] == 1 && data[1] == 1)
+        else
         {
-            int isGreen = 1;
-            for (int i = 2; i < 8; i++)
+            BlueToggleFlag = 0;
```

Comment:
I would say you don't need this else statement. We only want to update the toggle flag if the message has the blue led message.